Application.mk

```diff
@@ -93,6 +93,21 @@ ifneq ($(BUILD_MODULE),y)
   OBJS += $(MAINCOBJ) $(MAINCXXOBJ) $(MAINRUSTOBJ) $(MAINZIGOBJ)
 endif
 
+ifneq ($(PROGNAME),)
+  PROGOBJ := $(MAINCOBJ) $(MAINCXXOBJ) $(MAINRUSTOBJ)
+  PROGNAMELIST := $(PROGNAME)
+  ifneq ($(words $(PROGOBJ)), $(words $(PROGNAMELIST)))
+    $(warning "program names $(words $(PROGNAMELIST)) does not match mainsrcs $(words $(PROGOBJ)) in $(CWD)")
```

Comment:
why reference $(CWD)?

Reply:
> why reference $(CWD)?

**Removed**
I intend to throw a warning when there is a progname and mainsrc mismatch

Reply:
Yes, but we don't need `$(CWD)`.